client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java

```diff
@@ -191,46 +191,73 @@ public SchemaProviderImpl genSchemaProvider(long ver, Schema schema) {
                                     type.type,
                                     len,
                                     nullable,
-                                    hasDefault ? null : null);
+                                    hasDefault ? col.getDefault_value() : null);
         }
         return schemaProvider;
     }
 
     /**
      * @param spaceName the space name
-     * @param schemaName the tag name or edge name
+     * @param tagName the tag name
      * @param names the property names
      * @param values the property values
      * @return the encode byte[]
      * @throws RuntimeException expection
      */
     @Override
-    public byte[] encode(String spaceName,
-                         String schemaName,
-                         List<String> names,
-                         List<Object> values)
-        throws RuntimeException {
-        if (names == null || values == null || names.size() != values.size()) {
-            throw new RuntimeException("NebulaCodeImpl input wrong value");
+    public byte[] encodeTag(String spaceName,
+                            String tagName,
+                            List<String> names,
+                            List<Object> values) throws RuntimeException  {
+        TagItem tag = metaCache.getTag(spaceName, tagName);
+        if (tag == null) {
+            throw new RuntimeException(
```

Comment:
Exception should be more specific， such as IllegalArgumentException.

Reply:
it someday the getTag modify the implement, the tag return null, here will not modify, the interface here not need to depend on the details of `getTag`. check it is not null is more safely. the codec all expectations are using RuntimeException. and i think the tag is null is the RuntimeException, because it's not the expected exception.

Reply:
Here should throw RuntimeException. 
Just advice: RuntimeException is too abstract and it has many sub-exceptions such as NPE, IllegalArguementE, IndexOutOfBoundE,NoSuchElementE. The more specific the exception type, the easier to locate the problem.

Reply:
Yes, exception refinement is correct, but the exception here use RuntimeException is no problem.